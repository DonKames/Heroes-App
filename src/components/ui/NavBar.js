import React, { useContext } from 'react'
import { Link, NavLink, useNavigate } from 'react-router-dom'
import { AuthContext } from '../../auth/AuthContext';
import { types } from '../../types/types';

export const Navbar = () => {

    const navigate = useNavigate();

    const { user:{ name }, dispatch } = useContext(AuthContext);

    const handleLogout = () => {
        dispatch({
            type: types.logout
        });
        navigate('login', {replace:'true'});
    };


    return (
        <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
            
            <Link 
                className="navbar-brand" 
                to="./"
            >
                Asociaciones
            </Link>

            <div className="navbar-collapse">
                <div className="navbar-nav">

                    <NavLink 
                        activeclassname="active"
                        className="nav-item nav-link"
                        to="./marvel"
                    >
                        Marvel
                    </NavLink>

                    <NavLink 
                        activeclassname="active"
                        className="nav-item nav-link" 
                        to="./dc"
                    >
                        DC
                    </NavLink>

                    <NavLink 
                        activeclassname="active"
                        className="nav-item nav-link" 
                        to="./search"
                    >
                        Search
                    </NavLink>
                </div>
            </div>

            <div className="navbar-collapse collapse order-3 dual-collapse2">
                <ul className="navbar-nav ms-auto">
                    <span className='nav-item nav-link text-info'>
                        { name }
                    </span>
                    <button 
                        className="nav-item nav-link btn"
                        onClick={ handleLogout }
                    >
                        Logout
                    </button>
                </ul>
            </div>
        </nav>
    )
}