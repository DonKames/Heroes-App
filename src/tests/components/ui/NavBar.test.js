import { act } from '@testing-library/react';
import { mount } from 'enzyme';
import { MemoryRouter, Router } from 'react-router-dom';
import { AuthContext } from '../../../auth/AuthContext';
import { Navbar } from '../../../components/ui/NavBar';
import { types } from '../../../types/types';

describe('Pruebas en <NavBar />', () => {

  const navigateMock = jest.fn();

  const contextValue = {
    dispatch: jest.fn(),
    user: {
      name: 'trixie',
      logged: true
    }
  }

  const wrapper = mount(
    <AuthContext.Provider value={contextValue}>
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>
    </AuthContext.Provider>
  )


  test('Debe mostrarse correctamente.', () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('span').text()).toBe(contextValue.user.name);
  });

  
  test('Debe llamar el logout y el user history', () => {

    act(() => {
      wrapper.find('button').prop('onClick')();
    });
    
    expect(contextValue.dispatch).toHaveBeenCalledWith({ type: types.logout });


  });
});
