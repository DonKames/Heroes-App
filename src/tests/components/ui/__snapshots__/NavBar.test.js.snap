// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <NavBar /> Debe mostrarse correctamente. 1`] = `
<nav
  className="navbar navbar-expand-sm navbar-dark bg-dark"
>
  <a
    className="navbar-brand"
    href="/"
    onClick={[Function]}
  >
    Asociaciones
  </a>
  <div
    className="navbar-collapse"
  >
    <div
      className="navbar-nav"
    >
      <a
        activeclassname="active"
        className="nav-item nav-link"
        href="/marvel"
        onClick={[Function]}
      >
        Marvel
      </a>
      <a
        activeclassname="active"
        className="nav-item nav-link"
        href="/dc"
        onClick={[Function]}
      >
        DC
      </a>
      <a
        activeclassname="active"
        className="nav-item nav-link"
        href="/search"
        onClick={[Function]}
      >
        Search
      </a>
    </div>
  </div>
  <div
    className="navbar-collapse collapse order-3 dual-collapse2"
  >
    <ul
      className="navbar-nav ms-auto"
    >
      <span
        className="nav-item nav-link text-info"
      >
        trixie
      </span>
      <button
        className="nav-item nav-link btn"
        onClick={[Function]}
      >
        Logout
      </button>
    </ul>
  </div>
</nav>
`;
